### DISCUSSION TITLE
possible inconsistency of stream handling by rabbitmq on startup

### DISCUSSION THREAD

**Discussion by wrobell** [rabbitmq/rabbitmq-server#3873] [Other] [answered]
In both cases below, a RabbitMQ stream exists.

When starting RabbitMQ, my application can receive "Stream not available" (0x06) error when subscribing to a stream. I retry a second later and it works.

For the same testing scenario, when creating a publisher, I can get "Stream does not exists" (0x02) error.

This is very consistent for both cases, so far
- when subscribing I can get 0x06 and never 0x02
- when creating publisher I can get 0x02 and never 0x06

Would that be a bug? I would expect a consistent behaviour, this is to receive error 0x06 in both cases.

**Comment by acogoluegnes** [comment]
This is not intended, a function in the "declare publisher" command returns a result that covers both the "stream does not exist" and "stream not available" cases, we should refine this.

**Comment by acogoluegnes** [comment, marked as answer]
Follow-up issue: #3874.

**Reply by wrobell** [reply]
@acogoluegnes Thanks.
